components/motor/gpio/motor_encoder.go

```diff
@@ -256,13 +257,41 @@ func (m *EncodedMotor) goForMath(ctx context.Context, rpm, revolutions float64)
 // If revolutions is 0, this will run the motor at rpm indefinitely
 // If revolutions != 0, this will block until the number of revolutions has been completed or another operation comes in.
 func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
-	m.opMgr.CancelRunning(ctx)
 	ctx, done := m.opMgr.New(ctx)
 	defer done()
 
 	goalPos, goalRPM, direction := m.goForMath(ctx, rpm, revolutions)
 
-	return m.goForInternal(ctx, goalRPM, goalPos, direction)
+	if err := m.goForInternal(ctx, goalRPM, goalPos, direction); err != nil {
+		return err
+	}
+
+	if revolutions == 0 {
+		return nil
+	}
+
+	positionReached := func(ctx context.Context) (bool, error) {
+		var errs error
+		pos, posErr := m.position(ctx, extra)
+		errs = multierr.Combine(errs, posErr)
+		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
+			stopErr := m.Stop(ctx, extra)
+			errs = multierr.Combine(errs, stopErr)
+			return true, errs
+		}
+		return false, errs
+	}
+	err := m.opMgr.WaitForSuccess(
+		ctx,
+		10*time.Millisecond,
+		positionReached,
+	)
+	// Ignore the context canceled error - this occurs when the motor is stopped
+	// at the beginning of goForInternal
+	if !errors.Is(err, context.Canceled) {
+		return err
+	}
+	return nil
 }
 
 func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, goalPos, direction float64) error {
```

Comment:
Discussion question: `ctx` is now solely used for logging. Is it worth keeping in the function signature at all? I can go either way: tying the logs to a specific context is useful, but it strongly implies that `goForInternal` uses the context for something, which it frankly doesn't. I dunno; something to think about.

Reply:
sure I can remove it